### PR TITLE
Adding Protocol _Mapping for format_map method

### LINKED ISSUE #3824: format_map over specified
The [definition of format_map](https://github.com/python/typeshed/blob/92d53086be540c3eb6ead1967b32854689e4e7c3/stdlib/2and3/builtins.pyi#L436) indicates that the input is a `Mapping[str, Any]` when a `Mapping` is over-specified. `format_map` only requires the `__getitem__` interface of a `Mapping` (not `__iter__` nor `__len__`). Requiring `Mapping` forces someone writing an adapter to supply a full `Mapping` implementation for no purpose.

### PR DESCRIPTION
**Pull request by lladhibhutall**
closes #3824

### CODE REVIEW

File: stdlib/2/__builtin__.pyi [hunk 1]
```diff
@@ -26,17 +26,36 @@ _T = TypeVar('_T')
...
 _T3 = TypeVar('_T3')
 _T4 = TypeVar('_T4')
 _T5 = TypeVar('_T5')
 _TT = TypeVar('_TT', bound='type')
+_TC = TypeVar('_TC', bound=Type[object])
 
 class _SupportsIndex(Protocol):
     def __index__(self) -> int: ...
 
+def runtime_checkable(cls: _TC) -> _TC: ...
```
Comment (JelleZijlstra) [inline stdlib/2/__builtin__.pyi]: Why are you adding this (and Collections) to builtins? They don't exist in builtins at runtime.

Comment (lladhibhutall) [reply stdlib/2/__builtin__.pyi]: Thanks for correcting, I wasn't sure about this, the original stubs for Mapping had those.

Comment (JelleZijlstra) [reply stdlib/2/__builtin__.pyi:42]: Well if nothing else they should be imported from typing.

**Comment by JelleZijlstra** [comment]
I provided a simpler solution in #4122.

**Pull request closed without merging**